vscode/activation.rb

```diff
@@ -1,3 +1,3 @@
-env = ENV.map { |k, v| "#{k}RUBY_LSP_VS#{v}" }
+env = ENV.filter_map { |k, v| "#{k}RUBY_LSP_VS#{v}" unless v.encoding == Encoding::BINARY }
```

Comment:
Can we instead do:
```suggestion
env = ENV.filter_map { |k, v| "#{k}RUBY_LSP_VS#{v}" if v.force_encoding(Encoding::UTF_8).valid_encoding? }
```
this makes sure that we never try to encode any value that cannot be converted to UTF8, instead of special casing BINARY encoding here.

Reply:
Good idea. We don't care about binary specifically, just that it's UTF-8 compatible.